app/controllers/email_confirmations_controller.rb

```diff
@@ -0,0 +1,30 @@
+class EmailConfirmationsController < ApplicationController
+  before_action :validate_confirmation_token, only: :update
+
+  def update
+    @user.confirm_email!
+    sign_in @user
+    redirect_to root_path, notice: t('.confirmed_email')
+  end
+
+  def new
+  end
+
+  # used to resend confirmation mail for email validation
+  def create
+    user = User.find_by_email(params[:email_confirmation][:email])
+    if user
```

Comment:
Forgot to comment here :sweat_smile: I have incorporated only one part of requested change because rubocop would not have liked `if user = User.find...`.